### PR TITLE
[AKS] az aks check-acr: Add new check-acr command

### PR DESCRIPTION
**Pull request by samkreter**
**Description**

This PR add a new command `az aks check-acr` to the aks module. This command allows customers to easily check the ACR connection to the AKS cluster. See the testing guide for more details.

**Testing Guide**

Example usage: `az aks check-acr -g myResourceGroup -n myAKSCluster --acr myacr.azurecr.io`

Eample Output:

```
[2021-01-11T21:08:53Z] Checking host name resolution (myacr.azurecr.io): SUCCEEDED
[2021-01-11T21:08:53Z] Canonical name for ACR (myacr.azurecr.io): r1209eus.eastus.cloudapp.azure.com.
[2021-01-11T21:08:53Z] ACR location: eastus
[2021-01-11T21:08:53Z] Loading azure.json file from /etc/kubernetes/azure.json
[2021-01-11T21:08:53Z] The cluster uses service principal.
[2021-01-11T21:08:53Z] Validating service principal credential: SUCCEEDED
[2021-01-11T21:08:53Z] Validating image pull permission: SUCCEEDED
[2021-01-11T21:08:53Z]
Your cluster can pull images from myacr.azurecr.io!
```

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by yungezz** [comment]
hi @qwordy could you help to review?

**Comment by qwordy** [comment]
I downloaded the code and had a try. It seemed no problem. But, I haven't cover all code branches. Make sure you have tested all conditions.

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -1483,6 +1484,91 @@ def subnet_role_assignment_exists(cli_ctx, scope):
     return False
 
 
+def aks_check_acr(cmd, client, resource_group_name, name, acr):
+    if not which("kubectl"):
+        raise CLIError("Can not find kubectl executable in PATH")
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/acs/custom.py:1489]: CLIError is not recommended to use.
https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md

**Comment by qwordy** [comment]
@fengzhou-msft code owner

File: src/azure-cli/azure/cli/command_modules/acs/custom.py (hunk 1 above)
Comment (samkreter) [reply src/azure-cli/azure/cli/command_modules/acs/custom.py]: Update the error to a ValidationError

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**